pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -118,84 +118,113 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
+	flag := false
+	operatorConfig.Status.Conditions = nil
 	var availableConditionMessages []string
-
-	switch {
-	case actualDaemonSet == nil:
-		availableConditionReasons = append(availableConditionReasons, "NoDaemon")
-		availableConditionMessages = append(availableConditionMessages, "daemonset/apiserver.openshift-apiserver: could not be retrieved")
-	case actualDaemonSet.Status.NumberAvailable == 0:
-		availableConditionReasons = append(availableConditionReasons, "NoAPIServerPod")
-		availableConditionMessages = append(availableConditionMessages, "no openshift-apiserver daemon pods available on any node.")
-	case actualDaemonSet.Status.NumberAvailable > 0 && len(actualAPIServices) == 0:
-		availableConditionReasons = append(availableConditionReasons, "NoRegisteredAPIServices")
-		availableConditionMessages = append(availableConditionMessages, "registered apiservices could not be retrieved")
+	var availableConditionReasons []string
+	if actualDaemonSet == nil {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    daemonSetAvailable,
+			Status:  operatorv1.ConditionFalse,
+			Message: "daemonset/apiserver.openshift-apiserver: could not be retrieved",
+			Reason:  "NoDaemon",
+		})
+	} else if actualDaemonSet.Status.NumberAvailable == 0 {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    podAvailable,
+			Status:  operatorv1.ConditionFalse,
+			Message: "no openshift-apiserver daemon pods available on any node.",
+			Reason:  "NoAPIServerPod",
+		})
+	} else if actualDaemonSet.Status.NumberAvailable > 0 && len(actualAPIServices) == 0 {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    registeredAPIServiceAvailable,
+			Status:  operatorv1.ConditionFalse,
+			Message: "registered apiservices could not be retrieved",
+			Reason:  "NoRegisteredAPIServices",
+		})
+	} else {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:   operatorv1.OperatorStatusTypeAvailable,
+			Status: operatorv1.ConditionTrue,
+		})
 	}
+
 	for _, apiService := range actualAPIServices {
 		for _, condition := range apiService.Status.Conditions {
 			if condition.Type == apiregistrationv1.Available {
 				if condition.Status == apiregistrationv1.ConditionFalse {
 					availableConditionReasons = append(availableConditionReasons, "APIServiceNotAvailable")
 					availableConditionMessages = append(availableConditionMessages, fmt.Sprintf("apiservice/%v: not available: %v", apiService.Name, condition.Message))
+					v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+						Type:    apiServiceAvailable,
+						Status:  operatorv1.ConditionFalse,
+						Message: strings.Join(availableConditionMessages, "\n"),
+						Reason:  strings.Join(availableConditionReasons, "\n"),
```

Comment:
You seem to be creating one condition per failure, but concatenating the failure messages along the way. Please revisit this logic.